### PR TITLE
Update WindowDialog title when translation changes

### PR DESCRIPTION
**Pull request by YeldhamDev**

### CODE REVIEW

File: scene/gui/dialogs.cpp [hunk 1]
```diff
@@ -272,8 +282,11 @@ int WindowDialog::_drag_hit_test(const Point2 &pos) const {
 
 void WindowDialog::set_title(const String &p_title) {
 
-	title = tr(p_title);
-	update();
+	String new_title = tr(title);
```
Comment (merumelu) [inline scene/gui/dialogs.cpp]: Shouldn't this be `String new_title = tr(p_title);`?

Comment (YeldhamDev) [reply scene/gui/dialogs.cpp]: Oops, copy-paste error...

File: scene/gui/dialogs.cpp [hunk 2]
```diff
@@ -324,7 +337,7 @@ void WindowDialog::_bind_methods() {
 
 WindowDialog::WindowDialog() {
 
-	//title="Hello!";
+	title = "";
```
Comment (akien-mga) [inline scene/gui/dialogs.cpp]: `String` is default initialized to `""` already, so this is not necessary.

Comment (YeldhamDev) [reply scene/gui/dialogs.cpp]: Lots of strings are initialized like that throughout the codebase however, a cleanup may be needed.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2. (With follow-up #32440.)

**Pull request merged**